### ISSUE TITLE
two different relaunch buttons being displayed at times under `brave://settings/rewards`

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#19675]
## Description 

Sometimes two different type of `Relaunch` buttons will be used/displayed under `brave://settings/rewards` when enabling/displaying multiple settings under `Tip buttons`.

## Steps to Reproduce

   1. go into `brave://settings/rewards`
   2. enable/disable GH/Twitter/Reddit a few times (try a few combinations of the enabled/disabled at the same time)

You'll eventually run into an issue where you'll have two sets of different buttons

## Actual result:

![relaunchIssue](https://user-images.githubusercontent.com/2602313/142966508-babec4be-52a3-4e63-b5db-b1401548a173.png)

## Expected result:

![image](https://user-images.githubusercontent.com/2602313/142966530-0df705b0-4c4f-40ae-9fa9-dbc762195a5e.png)

## Reproduces how often: 

I would say there's ~35% or so change of reproducing. It doesn't always happen and I've only seen it a few times

## Brave version (brave://version info)

```
Brave | 1.34.28 Chromium: 96.0.4664.45 (Official Build) nightly (64-bit)
-- | --
Revision | 76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}
OS | Windows 11 Version 21H2 (Build 22000.318)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? `N/A`
- Can you reproduce this issue with the beta channel? `Yes` (once https://github.com/brave/brave-core/pull/11157 is uplifted)
- Can you reproduce this issue with the nightly channel? `Yes`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? `N/A`
- Does the issue resolve itself when disabling Brave Rewards? `N/A`
- Is the issue reproducible on the latest version of Chrome? `N/A`

## Miscellaneous Information:

CCing @Miyayes @LaurenWags @emerick

**Comment by emerick** [comment]
@kjozwiak Yes, I noticed this behavior as well. I'm wondering if it's an upstream bug with the button stying (though I can't find a ticket for it if it is), because I notice the same behavior with other settings like "Allow Google login for extensions" on brave://settings/extensions.

**Comment by emerick** [comment]
It seems like our Polymer overrides for `cr-button` don't work the second time you trigger them. Simple repro:

1. Visit brave://settings/extensions
2. Enable "Allow Google login for extensions" toggle
3. `Relaunch` button appears with appropriate styling
4. Disable "Allow Google login for extensions" toggle
5. `Relaunch` button disappears
6. Enable "Allow Google login for extensions" toggle
7. `Relaunch` button appears with inappropriate (default, I suppose) styling

@petemill Not sure if you might be familiar with what's happening here?

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

Brave | 1.45.75 Chromium: 106.0.5249.65 (Official Build) beta (x86_64)
-- | --
Revision | 3269dc3633cdd2ab94546fdbe54962e45b17a6e0-refs/branch-heads/5249@{#580}
OS | macOS Version 11.7 (Build 20G817)

Easily reproduced both issues using `1.44.101`

`brave://settings/rewards` | `brave://settings/extensions`
---------|---------
<img width="1312" alt="Screen Shot 2022-09-27 at 6 45 45 PM" src="https://user-images.githubusercontent.com/387249/192650946-dad023e0-ef69-448a-adc2-cc5808522bda.png"> | <img width="1182" alt="Screen Shot 2022-09-27 at 6 48 15 PM" src="https://user-images.githubusercontent.com/387249/192651112-44d06329-8a6e-40c6-a329-ba75c9246fe6.png">

### Steps:
1. installed `1.45.75`
2. launched Brave
3. opened `brave://settings/rewards`
4. enabled Rewards by clicking on `Brave Rewards are disabled`
5. click on `Start using Brave Rewards`
6. dismiss the tour
7. under `Tip buttons`, toggle `GitHub` to `Disabled`
8. confirmed a `Relaunch` button appears
9. toggled `Reddit` and `Twitter` back and forth a bunch of times

<img width="1312" alt="Screen Shot 2022-09-27 at 7 04 01 PM" src="https://user-images.githubusercontent.com/387249/192653265-f3ee2839-0eb8-4725-9775-68b2b8394b23.png">

Will verify pending resolution of what to do about the 2nd issue/instance from https://github.com/brave/brave-browser/issues/19675#issuecomment-976743105, here: https://bravesoftware.slack.com/archives/C7VLGSR55/p1664319694610449.
